sdk/servicebus/Azure.Messaging.ServiceBus/tests/RuleManager/RuleManagerLiveTests.cs

```diff
@@ -0,0 +1,588 @@
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual("SqlRuleUsingOperator", rulesDescription.First().Name);
+
+                await SendMessages(client, scope.TopicName);
+
+                var expectedOrders = Orders.Where(c => c.Color == "blue" || c.Quantity == 10).ToList();
+                await ReceiveMessages(
+                    client,
+                    scope.TopicName,
+                    scope.SubscriptionNames.First(),
+                    expectedOrders);
+            }
+        }
+
+        private Order[] Orders = new[]
+            {
+                new Order { Color = "blue", Quantity = 5, Priority = "low" },
+                new Order { Color = "red", Quantity = 10, Priority = "high" },
+                new Order { Color = "yellow", Quantity = 5, Priority = "low" },
+                new Order { Color = "blue", Quantity = 10, Priority = "low" },
+                new Order { Color = "blue", Quantity = 5, Priority = "high" },
+                new Order { Color = "blue", Quantity = 10, Priority = "low" },
+                new Order { Color = "red", Quantity = 5, Priority = "low" },
+                new Order { Color = "red", Quantity = 10, Priority = "low" },
+                new Order { Color = "red", Quantity = 5, Priority = "low" },
+                new Order { Color = "yellow", Quantity = 10, Priority = "high" },
+                new Order { Color = "yellow", Quantity = 5, Priority = "low" },
+                new Order { Color = "yellow", Quantity = 10, Priority = "low" }
+            };
+
+        private async Task SendMessages(ServiceBusClient client, string topicName)
+        {
+            ServiceBusSender sender = client.CreateSender(topicName);
+
+            for (int i = 0; i < Orders.Length; i++)
+            {
+                var message = new ServiceBusMessage(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(Orders[i])))
+                {
+                    CorrelationId = Orders[i].Priority,
+                    Label = Orders[i].Color,
+                    Properties =
+                {
+                    { "color", Orders[i].Color },
+                    { "quantity", Orders[i].Quantity },
+                    { "priority", Orders[i].Priority }
+                }
+                };
+                await sender.SendAsync(message);
+            }
+        }
+
+        private async Task ReceiveMessages(
+            ServiceBusClient client,
+            string topicName,
+            string subscriptionName,
+            IEnumerable<Order> expectedOrders)
+        {
+            var receiver = client.CreateReceiver(topicName, subscriptionName);
+            var messageEnum = expectedOrders.GetEnumerator();
+            var remainingMessages = expectedOrders.Count();
+            while (remainingMessages > 0)
+            {
+                foreach (var item in await receiver.ReceiveBatchAsync(Orders.Length).ConfigureAwait(false))
```

Comment:
nit: should this be receiveBatchAsync(remainingMessages)?

Reply:
No, here we are asking to receive all the orders but our expectation is to receive only the `expectedOrders` as per the applied filter. These tests are basically to check that we're receiving only expected messages as per the filter, even we ask for all the messages. That's how we test that the rules/filters are applied or not.